extensions/vtk/src/OHIFVTKViewport.js

```diff
@@ -146,46 +143,69 @@ class OHIFVTKViewport extends Component {
     studies,
     studyInstanceUid,
     displaySetInstanceUid,
-    sopClassUid,
     sopInstanceUid,
     frameIndex
   ) => {
     const stack = OHIFVTKViewport.getCornerstoneStack(
       studies,
       studyInstanceUid,
       displaySetInstanceUid,
-      sopClassUid,
       sopInstanceUid,
       frameIndex
     );
 
-    let imageDataObject;
+    const imageDataObject = getImageData(stack.imageIds, displaySetInstanceUid);
     let labelmapDataObject;
+    let labelmapColorLUT;
 
-    switch (sopClassUid) {
-      case SOP_CLASSES.SEGMENTATION_STORAGE:
-        throw new Error('Not yet implemented');
-      /*
-        const data = handleSegmentationStorage(
-          stack.imageIds,
-          displaySetInstanceUid
-        );
+    const firstImageId = stack.imageIds[0];
+    const { state } = segmentationModule;
+    const brushStackState = state.series[firstImageId];
+
+    if (brushStackState) {
+      const { activeLabelmapIndex } = brushStackState;
+      const labelmap3D = brushStackState.labelmaps3D[activeLabelmapIndex];
+
+      const vtkLabelmapID = `${firstImageId}_${activeLabelmapIndex}`;
+
+      if (labelmapCache[vtkLabelmapID]) {
+        labelmapDataObject = labelmapCache[vtkLabelmapID];
+      } else {
+        // TODO -> We need an imageId based getter in cornerstoneTools
+        const labelmapBuffer = labelmap3D.buffer;
 
-        imageDataObject = data.referenceDataObject;
-        labelmapDataObject = data.labelmapDataObject;
+        // Create VTK Image Data with buffer as input
+        labelmapDataObject = vtkImageData.newInstance();
 
-        return loadImageData(imageDataObject).then(() => {
-          return {
-            data: imageDataObject.vtkImageData,
-            labelmap: labelmapDataObject,
-          };
+        const dataArray = vtkDataArray.newInstance({
+          numberOfComponents: 1, // labelmap with single component
+          values: new Uint16Array(labelmapBuffer),
         });
-        */
-      default:
-        imageDataObject = getImageData(stack.imageIds, displaySetInstanceUid);
 
-        return imageDataObject;
+        labelmapDataObject.getPointData().setScalars(dataArray);
+        labelmapDataObject.setDimensions(...imageDataObject.dimensions);
+        labelmapDataObject.setSpacing(
+          ...imageDataObject.vtkImageData.getSpacing()
```

Comment:
What's in an imageDataObject?

Reply:
We are just using the same nomenclature as in `react-vtkjs-viewport`. This also has nothing do with this PR, I don't know why its highlighted as changed.